src/init.c

```diff
@@ -24,9 +24,15 @@
 #include <R_ext/Rdynload.h>
 #include "xxhash.h"
 #include "pmurhash.h"
+#include "digest.h"
+
+#define stringify(TAR) #TAR
+#define Register_ALGO(ALGO) R_RegisterCCallable("digest", stringify(direct_ ## ALGO), (DL_FUNC) &direct_ ## ALGO);
 
 void R_init_digest(DllInfo *info) {
     R_RegisterCCallable("digest", "PMurHash32", (DL_FUNC) &PMurHash32);
+    Register_ALGO(SHA1);
+    Register_ALGO(MD5);
```

Comment:
Not a huge fan _ex ante_ of `Register_ALGO`.  What `R_init_$PACKAGAME` does is pretty well understood and even if lining up multiple `R_RegisterCCallable()` looks tedious it is _simple_ and a good edtor can help filling in the stanza.  Every other package I have every seen writes it out, there is benefit in seeing the familar `R_RegisterCCallable()` rather than 'what is this ?' one has to chase to its definition to unmask.  And we probably won't expose a many dozen algos.

Reply:
fair - i'm mostly thinking of this as part of some machinery where the algos get enumerated in one place, then that machinery builds up the appropriate exports (among other things).

an alternative less-magic approach which ensures nothing missed would be some checker that confirms all the algos actually appear here.